src/Flysystem/S3/Tests/Integration/S3FilesystemV2Test.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AsyncAws\Flysystem\S3\Tests;
+
+use AsyncAws\Flysystem\S3\S3FilesystemV2;
+use AsyncAws\S3\S3Client;
+use League\Flysystem\Config;
+use League\Flysystem\FileAttributes;
+use League\Flysystem\FilesystemAdapter;
+use League\Flysystem\FilesystemAdapterTestCase;
+use League\Flysystem\StorageAttributes;
+
+class S3FilesystemV2Test extends FilesystemAdapterTestCase
+{
+    private $shouldCleanUp = false;
+
+    private static $adapterPrefix = 'test-prefix';
+
+    /**
+     * @var S3Client
+     */
+    private $s3Client;
+
+    public static function setUpBeforeClass(): void
+    {
+        static::$adapterPrefix = 'travis-ci/' . bin2hex(random_bytes(10));
+
+        if (!getenv('FLYSYSTEM_AWS_S3_KEY')) {
+            // Fixme, Docker is not working
+            self::markTestSkipped('Docker image is not working.');
```

Comment:
We dont need integration tests. They exist (or should exist) in the async-aws/s3 package.